*** /home/osm/openstreetmap-carto/roads.mss	2018-12-05 19:56:55.717147673 +0000
--- /home/osm/openstreetmap-carto/esdm/roads-OS.mss	2018-11-20 15:35:35.564785606 +0000
***************
*** 1,6 ****
  /* For the main linear features, such as roads and railways. */
  
! @tertiary-fill: #ffffff;
  @residential-fill: #ffffff;
  @service-fill: @residential-fill;
  @living-street-fill: #ededed;
--- 1,7 ----
  /* For the main linear features, such as roads and railways. */
  
! @tertiary-fill: #f7f607;
! @unclassified-fill: #f8f708;
  @residential-fill: #ffffff;
  @service-fill: @residential-fill;
  @living-street-fill: #ededed;
***************
*** 24,36 ****
  @access-marking: #eaeaea;
  @access-marking-living-street: #cccccc;
  
! @default-casing: white;
! @tertiary-casing: #8f8f8f;
! @residential-casing: #bbb;
  @road-casing: @residential-casing;
  @service-casing: @residential-casing;
  @living-street-casing: @residential-casing;
! @pedestrian-casing: #999;
  @path-casing: @default-casing;
  @footway-casing: @default-casing;
  @steps-casing: @default-casing;
--- 25,38 ----
  @access-marking: #eaeaea;
  @access-marking-living-street: #cccccc;
  
! @default-casing: #000000;
! @tertiary-casing: #000000;
! @unclassified-casing: #000000;
! @residential-casing: #101010;
  @road-casing: @residential-casing;
  @service-casing: @residential-casing;
  @living-street-casing: @residential-casing;
! @pedestrian-casing: #404040;
  @path-casing: @default-casing;
  @footway-casing: @default-casing;
  @steps-casing: @default-casing;
***************
*** 40,46 ****
  
  @tertiary-shield: #3b3b3b;
  
! @unimportant-road: @residential-casing;
  
  @residential-construction: #aaa;
  @service-construction: #aaa;
--- 42,48 ----
  
  @tertiary-shield: #3b3b3b;
  
! @unimportant-road: @unclassified-fill;
  
  @residential-construction: #aaa;
  @service-construction: #aaa;
***************
*** 58,63 ****
--- 60,66 ----
  @secondary-tunnel-fill: lighten(@secondary-fill, 5%);
  @tertiary-tunnel-fill: lighten(@tertiary-fill, 5%);
  @residential-tunnel-fill: darken(@residential-fill, 5%);
+ @unclassified-tunnel-fill: darken(@unclassified-fill, 5%);
  @living-street-tunnel-fill: lighten(@living-street-fill, 3%);
  
  @motorway-width-z6:               0.4;
***************
*** 95,108 ****
  @primary-link-width-z12:          1.5;
  @secondary-width-z12:             3.5;
  @secondary-link-width-z12:        1.5;
! @tertiary-width-z12:              2.5;
  @residential-width-z12:           0.5;
! @unclassified-width-z12:          0.8;
  
  @motorway-width-z13:              6;
  @motorway-link-width-z13:         4;
  @trunk-width-z13:                 6;
! @trunk-link-width-z13:             4;
  @primary-width-z13:               5;
  @primary-link-width-z13:          4;
  @secondary-width-z13:             5;
--- 98,111 ----
  @primary-link-width-z12:          1.5;
  @secondary-width-z12:             3.5;
  @secondary-link-width-z12:        1.5;
! @tertiary-width-z12:              3.0;
  @residential-width-z12:           0.5;
! @unclassified-width-z12:          2.5;
  
  @motorway-width-z13:              6;
  @motorway-link-width-z13:         4;
  @trunk-width-z13:                 6;
! @trunk-link-width-z13:            4;
  @primary-width-z13:               5;
  @primary-link-width-z13:          4;
  @secondary-width-z13:             5;
***************
*** 215,220 ****
--- 218,224 ----
  @major-casing-width-z11:          0.3;
  
  @casing-width-z12:                0.3;
+ @unclassified-casing-width-z12:   0.3;
  @secondary-casing-width-z12:      0.3;
  @major-casing-width-z12:          0.5;
  
***************
*** 302,310 ****
  @shield-font: @book-fonts;
  @shield-clip: false;
  
- @major-highway-text-repeat-distance: 50;
- @minor-highway-text-repeat-distance: 10;
- 
  @railway-text-repeat-distance: 200;
  
  #roads-casing, #bridges, #tunnels {
--- 306,311 ----
***************
*** 476,483 ****
        }
      }
  
!     [feature = 'highway_residential'],
!     [feature = 'highway_unclassified'] {
        [zoom >= 13] {
          line-color: @residential-casing;
          line-width: @residential-width-z13;
--- 477,483 ----
        }
      }
  
!     [feature = 'highway_residential'] {
        [zoom >= 13] {
          line-color: @residential-casing;
          line-width: @residential-width-z13;
***************
*** 502,507 ****
--- 502,534 ----
          }
        }
      }
+     
+     [feature = 'highway_unclassified'] {
+       [zoom >= 12] {
+         line-color: @unclassified-casing;
+         line-width: @unclassified-width-z12;
+         [zoom >= 13] { line-width: @residential-width-z13; }
+         [zoom >= 14] { line-width: @residential-width-z14; }
+         [zoom >= 15] { line-width: @residential-width-z15; }
+         [zoom >= 16] { line-width: @residential-width-z16; }
+         [zoom >= 17] { line-width: @residential-width-z17; }
+         [zoom >= 18] { line-width: @residential-width-z18; }
+         [zoom >= 19] { line-width: @residential-width-z19; }
+         #roads-casing {
+           line-join: round;
+           line-cap: round;
+         }
+         #tunnels {
+           line-dasharray: 4,2;
+         }
+         #bridges {
+           [zoom >= 14] {
+             line-color: @bridge-casing;
+             line-join: round;
+           }
+         }
+       }
+     }
  
      [feature = 'highway_road'] {
        [zoom >= 14] {
***************
*** 1164,1173 ****
          [construction = 'secondary_link'] {
            line-color: @secondary-fill;
          }
          [construction = 'tertiary'],
          [construction = 'tertiary_link'],
          [construction = 'residential'],
-         [construction = 'unclassified'],
          [construction = 'living_street'] {
            line-color: @residential-construction;
            [zoom < 13] {
--- 1191,1202 ----
          [construction = 'secondary_link'] {
            line-color: @secondary-fill;
          }
+         [construction = 'unclassified'] {
+ 					line-color: @unclassified-fill
+ 				}
          [construction = 'tertiary'],
          [construction = 'tertiary_link'],
          [construction = 'residential'],
          [construction = 'living_street'] {
            line-color: @residential-construction;
            [zoom < 13] {
***************
*** 1464,1477 ****
      [feature = 'highway_residential'],
      [feature = 'highway_unclassified'] {
        [zoom = 12][feature = 'highway_residential'] {
!         line-color: @unimportant-road;
          line-width: @residential-width-z12;
        }
        [zoom = 12][feature = 'highway_unclassified'] {
!         line-color: @unimportant-road;
!         line-width: @unclassified-width-z12;
        }
!       [zoom >= 13] {
          line-width: @residential-width-z13 - 2 * @residential-casing-width-z13;
          [zoom >= 14] { line-width: @residential-width-z14 - 2 * @casing-width-z14; }
          [zoom >= 15] { line-width: @residential-width-z15 - 2 * @casing-width-z15; }
--- 1493,1506 ----
      [feature = 'highway_residential'],
      [feature = 'highway_unclassified'] {
        [zoom = 12][feature = 'highway_residential'] {
!         line-color: @residential-fill;
          line-width: @residential-width-z12;
        }
        [zoom = 12][feature = 'highway_unclassified'] {
!         line-color: @unclassified-fill;
!         line-width: @unclassified-width-z12 - 2 * @unclassified-casing-width-z12;
        }
!       [zoom >= 13][feature = 'highway_residential'] {
          line-width: @residential-width-z13 - 2 * @residential-casing-width-z13;
          [zoom >= 14] { line-width: @residential-width-z14 - 2 * @casing-width-z14; }
          [zoom >= 15] { line-width: @residential-width-z15 - 2 * @casing-width-z15; }
***************
*** 1497,1502 ****
--- 1526,1557 ----
          line-cap: round;
          line-join: round;
        }
+       [zoom >= 13][feature = 'highway_unclassified'] {
+         line-width: @residential-width-z13 - 2 * @residential-casing-width-z13;
+         [zoom >= 14] { line-width: @residential-width-z14 - 2 * @casing-width-z14; }
+         [zoom >= 15] { line-width: @residential-width-z15 - 2 * @casing-width-z15; }
+         [zoom >= 16] { line-width: @residential-width-z16 - 2 * @casing-width-z16; }
+         [zoom >= 17] { line-width: @residential-width-z17 - 2 * @casing-width-z17; }
+         [zoom >= 18] { line-width: @residential-width-z18 - 2 * @casing-width-z18; }
+         [zoom >= 19] { line-width: @residential-width-z19 - 2 * @casing-width-z19; }
+         #roads-fill, #bridges {
+           line-color: @unclassified-fill;
+         }
+         #tunnels {
+           line-color: @unclassified-tunnel-fill;
+         }
+         #bridges {
+           line-width: @residential-width-z13 - 2 * @bridge-casing-width-z13;
+           [zoom >= 14] { line-width: @residential-width-z14 - 2 * @bridge-casing-width-z14; }
+           [zoom >= 15] { line-width: @residential-width-z15 - 2 * @bridge-casing-width-z15; }
+           [zoom >= 16] { line-width: @residential-width-z16 - 2 * @bridge-casing-width-z16; }
+           [zoom >= 17] { line-width: @residential-width-z17 - 2 * @bridge-casing-width-z17; }
+           [zoom >= 18] { line-width: @residential-width-z18 - 2 * @bridge-casing-width-z18; }
+           [zoom >= 19] { line-width: @residential-width-z19 - 2 * @bridge-casing-width-z19; }
+         }
+         line-cap: round;
+         line-join: round;
+       }
      }
  
      [feature = 'highway_living_street'] {
***************
*** 1921,1928 ****
      [feature = 'railway_rail'][zoom >= 8],
      [feature = 'railway_INT-spur-siding-yard'][zoom >= 13] {
        [zoom < 13] {
!         line-color: #787878;
!         line-width: 0.5;
          [zoom >= 8] { line-width: 0.8; }
          [zoom >= 12] { line-width: 0.9; }
          line-join: round;
--- 1976,1983 ----
      [feature = 'railway_rail'][zoom >= 8],
      [feature = 'railway_INT-spur-siding-yard'][zoom >= 13] {
        [zoom < 13] {
!         line-color: #585858;
!         line-width: 0.7;
          [zoom >= 8] { line-width: 0.8; }
          [zoom >= 12] { line-width: 0.9; }
          line-join: round;
***************
*** 2894,2903 ****
        text-spacing: 750;
        text-clip: false;
        text-placement: line;
        text-face-name: @book-fonts;
        text-halo-radius: @standard-halo-radius;
        text-halo-fill: @standard-halo-fill;
-       text-repeat-distance: @minor-highway-text-repeat-distance;
      }
    }
  }
--- 2949,2958 ----
        text-spacing: 750;
        text-clip: false;
        text-placement: line;
+       text-min-distance: 18;
        text-face-name: @book-fonts;
        text-halo-radius: @standard-halo-radius;
        text-halo-fill: @standard-halo-fill;
      }
    }
  }
***************
*** 2917,2923 ****
        text-clip: false;
        text-placement: line;
        text-face-name: @book-fonts;
-       text-repeat-distance: @major-highway-text-repeat-distance;
        [tunnel = 'no'] {
          text-halo-radius: @standard-halo-radius;
          [highway = 'motorway'] { text-halo-fill: @motorway-fill; }
--- 2972,2977 ----
***************
*** 2950,2956 ****
        text-face-name: @book-fonts;
        text-halo-radius: @standard-halo-radius;
        text-halo-fill: @secondary-fill;
-       text-repeat-distance: @major-highway-text-repeat-distance;
      }
      [zoom >= 14] {
        text-size: 9;
--- 3004,3009 ----
***************
*** 2977,2983 ****
        text-face-name: @book-fonts;
        text-halo-radius: @standard-halo-radius;
        text-halo-fill: @tertiary-fill;
-       text-repeat-distance: @major-highway-text-repeat-distance;
      }
      [zoom >= 17] {
        text-size: 11;
--- 3030,3035 ----
***************
*** 2996,3002 ****
      text-halo-radius: @standard-halo-radius;
      text-halo-fill: @standard-halo-fill;
      text-face-name: @book-fonts;
-     text-repeat-distance: @major-highway-text-repeat-distance;
  
      [zoom >= 17] {
        text-size: 11;
--- 3048,3053 ----
***************
*** 3023,3030 ****
        text-halo-radius: @standard-halo-radius;
        text-halo-fill: @residential-fill;
        text-face-name: @book-fonts;
-       text-repeat-distance: @minor-highway-text-repeat-distance;
-       [highway = 'unclassified'] { text-repeat-distance: @major-highway-text-repeat-distance;}
      }
      [zoom >= 16] {
        text-size: 9;
--- 3074,3079 ----
***************
*** 3054,3060 ****
        [highway = 'raceway'] { text-halo-fill: @raceway-fill; }
        [highway = 'service'] { text-halo-fill: @service-fill; }
        text-face-name: @book-fonts;
-       text-repeat-distance: @major-highway-text-repeat-distance;
      }
      [zoom >= 17] {
        text-size: 11;
--- 3103,3108 ----
***************
*** 3073,3085 ****
        text-clip: false;
        text-placement: line;
        text-halo-radius: @standard-halo-radius;
!       [highway = 'living_street'] { 
!         text-halo-fill: @living-street-fill;
!         text-repeat-distance: @major-highway-text-repeat-distance;
!       }
        [highway = 'pedestrian'] { text-halo-fill: @pedestrian-fill; }
        text-face-name: @book-fonts;
-       text-repeat-distance: @minor-highway-text-repeat-distance;
      }
      [zoom >= 16] {
        text-size: 9;
--- 3121,3129 ----
        text-clip: false;
        text-placement: line;
        text-halo-radius: @standard-halo-radius;
!       [highway = 'living_street'] { text-halo-fill: @living-street-fill; }
        [highway = 'pedestrian'] { text-halo-fill: @pedestrian-fill; }
        text-face-name: @book-fonts;
      }
      [zoom >= 16] {
        text-size: 9;
***************
*** 3119,3125 ****
        text-face-name: @book-fonts;
        text-vertical-alignment: middle;
        text-dy: 5;
-       text-repeat-distance: @major-highway-text-repeat-distance;
      }
      [zoom >= 16] {
        text-size: 9;
--- 3163,3168 ----
***************
*** 3148,3155 ****
        text-face-name: @book-fonts;
        text-vertical-alignment: middle;
        text-dy: 7;
-       text-repeat-distance: @major-highway-text-repeat-distance;
-       [highway = 'steps'] { text-repeat-distance: @minor-highway-text-repeat-distance; }
      }
      [zoom >= 17] {
        text-size: 11;
--- 3191,3196 ----
